Engine/src/Engine/ThreadPool.cpp

```diff
@@ -0,0 +1,32 @@
+#include "ThreadPool.h"
+#include <iostream>
+#include <thread>
+#include "Log.h"
+
+using namespace engine_concurrent;
+
+ThreadPool* ThreadPool::tpInstance = nullptr;
+std::mutex ThreadPool::mutex; // Instantiate the mutex
+
+// THE JOB OF THREAD POOL IS JUST TO MAKE ALL THE THREADS and maybe more things later 
+
+// Constructor for thread pool, is run once at the start and shouldn't be run again
+//make 1 thread for each core, -1 to leave space for the main thread
+ThreadPool::ThreadPool() : m_threads(std::thread::hardware_concurrency()-1)
+{
+    for (int i = 0; i < m_threads.size(); i++)
+    {
+        m_threads[i] = new Thread(i);

```

Comment:
I don't see Thread getting deleted. Possible memory leak?

Reply:
Also is there a reason to call "new" for the wrapper class here? Can this be created on the stack?

Reply:
Fixed: added d'tor